File: solutions/typescript/grade-school/1/grade-school.ts
export class GradeSchool {
  private schoolRoster: { [grade: number]: string[] } = {};
  roster() {
    const sortedRoster: { [grade: number]: string[] } = {};
    const grades = Object.keys(this.schoolRoster).map(Number).sort((a, b) => a - b);
    for (const grade of grades) {
      sortedRoster[grade] = [...this.schoolRoster[grade]].sort();
    }
    return JSON.parse(JSON.stringify(sortedRoster));
  }

  add(name: string, grade: number) {
    for (const g in this.schoolRoster) {
      this.schoolRoster[g] = this.schoolRoster[g].filter(student => student !== name);
    }
    if (!this.schoolRoster[grade]) {
      this.schoolRoster[grade] = [];
    }
    this.schoolRoster[grade].push(name);
  }

  grade(grade: number) {
    return this.schoolRoster[grade] ? [...this.schoolRoster[grade]].sort() : [];
  }
}

File: solutions/kotlin/grade-school/1/src/main/kotlin/GradeSchool.kt
class School {
 
    private val students = mutableMapOf<String, Int>()

    fun add(student: String, grade: Int) {
        students[student] = grade
    }

    fun grade(grade: Int): List<String> {
        return students.filter { it.value == grade }
            .keys
            .sorted()
    }

    fun roster(): List<String> {
        return students.entries
            .sortedWith(compareBy<Map.Entry<String, Int>> { it.value }.thenBy { it.key })
            .map { it.key }
    }
}
